lib/Dialect/TritonGPU/Transforms/Utility.cpp

```diff
@@ -668,7 +668,15 @@ getConvertBackwardSlice(Value root, SetVector<Value> &slice,
                         Attribute rootEncoding,
                         DenseMap<Value, Attribute> &layout,
                         std::function<bool(Operation *)> stopPropagation) {
+  DenseSet<Value> visited;
   SmallVector<std::pair<Value, Attribute>> queue = {{root, rootEncoding}};
+  auto enqueueIfNew = [&](Value v, Attribute attr) {
+    if (visited.contains(v)) {
+      return;
+    }
+    visited.insert(v);
+    queue.push_back({v, attr});
+  };
   while (!queue.empty()) {
     auto [currentValue, encoding] = queue.back();
```

Comment:
Can we simplify this a little by doing:
```
DenseSet<Value> visited;
while (!queue.empty()) {
  auto [currentValue, encoding] = queue.back();
  if (!visited.insert(currentValue).second)
    continue;
```
This way no need for the lambda helper and it should be same efficiency.

Reply:
Yes, even better:)